### DISCUSSION TITLE
How to run c/c++ on code-server

### DISCUSSION THREAD

**Discussion by rlarudqls630** [coder/code-server#4606] [General]
Hi
I am currently running code-server on ubuntu inside windows 10 using fairy new feature of windows 10.

In this setup I was not able to compile c/c++.
However, running java and python on the server is no problem.

It is giving me an error saying that I need to download the c/c++ compiler, but it seems like vscode original extension is not an option.

Can someone please help me to compile c/c++ on code-server.

Thank you.

**Comment by lg-nineoneninethreenine** [comment]
You need the `gcc` and `g++` packages(`sudo apt update && sudo apt install 
gcc g++ -y`) and then use the GCC compiler.

**Reply by rlarudqls630** [reply]
How do i use gcc compiler?
Never used that before.
I am pretty new to c/c++ programing.
Thank you

**Comment by lg-nineoneninethreenine** [comment]
I kinda think it's like this:
```bash
$ gcc file.c file
```
You can try a Google search to check it out or since you are on Windows, 
you can get the Turbo C Compiler. But it has one issue, it locks your 
cursor like these games like Minecraft, the only way to break out of it is 
to use the <kbd>Alt</kbd>+<kbd>Tab</kbd> method to free your cursor.

**Reply by rlarudqls630** [reply]
Thank you 
I'll try!!

**Reply by rlarudqls630** [reply]
I tried it and it worked but output file is broken.

It says unsupported text encoding.
Do I need another extension to see the output?

Sorry for bothering.

**Comment by lg-nineoneninethreenine** [comment]
Nah, try the same code on Turbo C Compiler on Windows, on Linux, some 
imports failed, like the CONIO always failed, but on Turbo C it works.

**Reply by rlarudqls630** [reply]
Thank you.
I found other way to see the output!!
I used
$./a.out

**Reply by lg-nineoneninethreenine** [reply]
Ah that's the way, thought I missed something.
